### PR TITLE
Module 2 (One-dimensional arrays)

### PR DESCRIPTION
**Pull request by pmiakish**

### CODE REVIEW

File: module_2/src/com/epam/onedimensionalarrays/SumOfNumbersDivisibleByK.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            int sum = 0;
+            for (int i : integerArray) {
+                if ((i % k) == 0) {
+                    sum += i;
+                }
+            }
+            System.out.print(sum);
+        } else {
+            System.out.print("Incorrect values! try again...");
+        }
```
[the remark this answers is not in the document]
Comment (pmiakish) [reply module_2/src/com/epam/onedimensionalarrays/SumOfNumbersDivisibleByK.java:34]: I don't quite catch why int sum should be assigned a value -1

File: module_2/src/com/epam/onedimensionalarrays/EachSecondElementOfArrayCompressor.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * The class for optimizing integer array elements
+ * (compression of each second element, filling emptied elements with zeros).
+ */;
+
+public class EachSecondElementOfArrayCompressor {
+
+    private static int currentWritingIndex = 0;
+
+    public static int[] compressArray(int[] generatedNumbers) {
+        generatedNumbers = rewriteElementsWithEvenIndexes(generatedNumbers);
```
Comment (lovelyserafima) [inline module_2/src/com/epam/onedimensionalarrays/EachSecondElementOfArrayCompressor.java:14]: just void method, don't need to return it

Comment (lovelyserafima) [reply module_2/src/com/epam/onedimensionalarrays/EachSecondElementOfArrayCompressor.java:14]: I mean rewrtiteBlaBla

File: module_2/src/com/epam/onedimensionalarrays/EachSecondElementOfArrayCompressor.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+/**
+ * Task #10:
+ * The class for optimizing integer array elements
+ * (compression of each second element, filling emptied elements with zeros).
+ */;
+
+public class EachSecondElementOfArrayCompressor {
+
+    private static int currentWritingIndex = 0;
```
Comment (lovelyserafima) [inline module_2/src/com/epam/onedimensionalarrays/EachSecondElementOfArrayCompressor.java:11]: static

File: module_2/src/com/epam/onedimensionalarrays/EachSecondElementOfArrayCompressor.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        generatedNumbers = rewriteElementsWithEvenIndexes(generatedNumbers);
+        for (int i = currentWritingIndex; i < generatedNumbers.length; i++) {
+            generatedNumbers[i] = 0;
+        }
+        return generatedNumbers;
+    }
+
+    private static int[] rewriteElementsWithEvenIndexes(int[] generatedNumbers) {
+        for (int currentReadingIndex = 0; currentReadingIndex < generatedNumbers.length; currentReadingIndex++) {
+            if (currentReadingIndex % 2 == 0) {
```
Comment (lovelyserafima) [inline module_2/src/com/epam/onedimensionalarrays/EachSecondElementOfArrayCompressor.java:23]: You can simplify it like:
```suggestion
        for (int currentReadingIndex = 0; currentReadingIndex < generatedNumbers.length; currentReadingIndex += 2) {
```

File: module_2/src/com/epam/onedimensionalarrays/EachSecondElementOfArrayCompressor.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+public class EachSecondElementOfArrayCompressor {
+
+    private static int currentWritingIndex = 0;
+
+    public static int[] compressArray(int[] generatedNumbers) {
+        generatedNumbers = rewriteElementsWithEvenIndexes(generatedNumbers);
+        for (int i = currentWritingIndex; i < generatedNumbers.length; i++) {
+            generatedNumbers[i] = 0;
+        }
+        return generatedNumbers;
```
Comment (lovelyserafima) [inline module_2/src/com/epam/onedimensionalarrays/EachSecondElementOfArrayCompressor.java:18]: E.g. you can start not from WritingIndex, just half of the array

File: module_2/src/com/epam/onedimensionalarrays/TaskLauncher.java [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+        // Tasks
+        switch (taskNumber) {
+            case 1:
+                int[] generatedIntNumbersForSumming = ArrayMaker.generateArrayOfIntNumbers(MIN_VALUE_FIRST_TASK, MAX_VALUE_FIRST_TASK, AMOUNT_OF_NUMBERS_FIRST_TASK);
+                System.out.println("\nThe generated array:\n" + Arrays.toString(generatedIntNumbersForSumming));
+                int sum = SumCalculatorOfNumbersDivisibleByK.countSum(generatedIntNumbersForSumming, DIVISOR_K);
+                if (sum > 0) {
+                    System.out.println("\nThe sum of the numbers divisible by K is " + sum);
+                }
```
[the remark this answers is not in the document]
Comment (pmiakish) [reply module_2/src/com/epam/onedimensionalarrays/TaskLauncher.java:68]: I thought that the message "Incorrect divisor value" in method countSum() was enough

[the remark this answers is not in the document]
Comment (lovelyserafima) [reply module_2/src/com/epam/onedimensionalarrays/TaskLauncher.java:71]: Sorry, I didn't see it:)

**Pull request merged**